### ISSUE TITLE
Style changes merge bottleneck

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#6167]
Its great that we have more people contributing to styles lately! Many thanks to everyone involved! @RedAuburn @dvdmrtnz @map-per

However it made apparent that we have a shortcoming with our workflow:
there is a queue of style PRs waiting to be merged, but we can't merge them at once because after one is merged all other need to be rebased, which takes time and focus unnecessarily.

Merge conflicts are due to binary `drules_*.bin` files (and `symbols.png` skin files when icons are changes, but its not that big issue *for now*) which just can't be merged.

Also complete rewrite of binary files for every minor change makes our repository size grow bigger faster and faster (e.g. a single icon update leads to ~5Mb of additional skin files versions).
Also each rebase and push makes the CI builds and tests run anew (takes time and there is a rate limit for free github CI, right?).

An ultimate solution would be to stop storing binary drules and skin files in the repo, but generate them on-the-fly from the sources e.g. just before building an app.

However there are many dependencies on those binaries ATM which make it a big task:
- style binaries should be generated automatically/transparently for each platform build, so that a build env setup process is not more complicated
- automatic tests run by github CI (which include styles-related tests) depend on style binaries too, so they have to be generated automatically there too (and it shouldn't slow down this process much)
- similarly, the maps generator depends on style binaries too

So it would be great to find an interim simpler solution, let's brainstorm!

CC @biodranik @vng

**Comment by pastk** [comment]
One option is:
- keep the style binaries in the repo (so no need to update platform build toolchanins), but don't push/update in each PR
- instead, a dev with master merge rights will merge several style PRs and re-generate binaries once
- but this way we'll loose automatic CI tests benefit, so
    - update CI tests to re-generate binaries when necessary (better, but probably harder)
    - run those tests on contributor's machines before pushing changes (e.g. add a `generate_drules.sh --commit` option that'll run the tests but will remove style binaries from a final commit)

Any other ideas?

**Comment by map-per** [comment]
In the short term adding `@biodranik @vng` when rebasing could help to prevent the rebase from being overlooked

**Comment by biodranik** [comment]
Yes, if PR is LGTM to everyone and polished enough, we can merge it quickly (ping if necessary).

**Comment by dvdmrtnz** [comment]
What about dropping the Regeneration commit before doing the merge? Can the merger do this? Then after some style changes have accumulated (or right before the compilation for release/beta) a regeneration could be done.

Then there would be no conflicts in binary files when merging, and there would be no pointless increase in the file size of the repository.

**Comment by pastk** [comment]
> What about dropping the Regeneration commit before doing the merge? Can the merger do this? Then after some style changes have accumulated (or right before the compilation for release/beta) a regeneration commit could be done.

It looks like there is no UI in Github to do it easily.
It should be possible to do it by adding contributor's repo as a remote, pulling the pr branch from it, removing the commit and force pushing it back into contributor's repo/branch (there is a checkbox for a contributor to allow maintainers to do it, its ON by default).

But as you can see its a lot of work for a maintainer...

So an easier option would be for contributors themselves to drop the last commit after getting a merge approval.

WDYT?

**Comment by pastk** [comment]
A more radical idea:
- https://github.com/organicmaps/organicmaps/issues/10341
